### ISSUE TITLE
Add debugging messages and configurable verbosity

### ISSUE THREAD

**Issue by jameslamb** [rapidsai/rapids-build-backend#24]
While working through https://github.com/rapidsai/rmm/pull/1502, I've found myself struggling to debug problems of the form *"`rapids-build-backend` isn't doing what I want, but also isn't raising any warnings or errors"*.

It'd be helpful to be able to turn on debug-level logging and see exactly what the library is doing.

For example, I'd like to be able to use logs to inspect what's happening in here:

https://github.com/rapidsai/rapids-build-backend/blob/bbe5b13ea13bebd875dfed7d4514b34b768c3927/rapids_build_backend/impls.py#L180-L205

### References

Some related conversations:

* https://github.com/pypa/pip/issues/6634
* https://github.com/scikit-build/scikit-build-core/pull/513

**Comment by vyasr** [comment]
One way to do this would be to use the [`logging` module](https://docs.python.org/3/library/logging.html) and configure log levels using an environment variable. It's tricky because the build frontend has most of the control of how the backend is invoked, so you really have to use an environment variable or something similar to control the log levels because you can't force any information to be passed via the CLI/API.

**Comment by jameslamb** [comment]
> *so you really have to use an environment variable or something similar to control the log levels*

`rapids-build-backend` could expose configuration for logging verbosity that's passed through config-settings from `pyproject.toml`, just like `scikit-build-core` does with `cmake.verbose`.

like this

```toml
[tool.rapids-build-backend]
log-level = "DEBUG"
```

**Comment by vyasr** [comment]
Yup that would work. RBB supports configuration, and ~all config variables in pyproject.toml may be overridden with config settings or an env var.
